yarn-project/cli/src/config/chain_l2_config.ts

```diff
@@ -265,37 +262,13 @@ export const testnetL2ChainConfig: L2ChainConfig = {
 const BOOTNODE_CACHE_DURATION_MS = 60 * 60 * 1000; // 1 hour;
 
 export async function getBootnodes(networkName: NetworkNames, cacheDir?: string) {
-  const cacheFile = cacheDir ? join(cacheDir, networkName, 'bootnodes.json') : undefined;
-  try {
-    if (cacheFile) {
-      const info = await stat(cacheFile);
-      if (info.mtimeMs + BOOTNODE_CACHE_DURATION_MS > Date.now()) {
-        return JSON.parse(await readFile(cacheFile, 'utf-8'))['bootnodes'];
-      }
-    }
-  } catch {
-    // no-op. Get the remote-file
-  }
-
   const url = `http://static.aztec.network/${networkName}/bootnodes.json`;
-  const response = await fetch(url);
-  if (!response.ok) {
-    throw new Error(
-      `Failed to fetch basic contract addresses from ${url}. Check you are using a correct network name.`,
-    );
-  }
-  const json = await response.json();
-
-  try {
-    if (cacheFile) {
-      await mkdir(dirname(cacheFile), { recursive: true });
-      await writeFile(cacheFile, JSON.stringify(json), 'utf-8');
-    }
-  } catch {
-    // no-op
-  }
+  const data = await cachedFetch(url, {
+    cacheDurationMs: BOOTNODE_CACHE_DURATION_MS,
+    cacheFile: cacheDir ? join(cacheDir, networkName, 'bootnodes.json') : undefined,
+  });
 
-  return json['bootnodes'];
+  return data.bootnodes;
 }
```

Comment:
We no longer crash if the bootnodes file is not available (fix A-61), so I've added a warning if starting a P2P client with 0 boot nodes defined.